code/game/machinery/doors/door.dm

```diff
@@ -516,6 +517,22 @@
 		qdel(fire)
 	return
 
+/obj/machinery/door/proc/crush(var/damage)
+	var/damagetodo = damage ? damage : DOOR_CRUSH_DAMAGE
+	for(var/mob/living/L in get_turf(src))
+		if(isalien(L))  //For xenos

```

Comment:
This will trigger on nymphs and larvae, but not full size xenos. They'll be caught by the statement below.